addons/api/addon/transforms/object-array.js

```diff
@@ -0,0 +1,18 @@
+import Transform from '@ember-data/serializer/transform';
+
+export default class ObjectArrayTransform extends Transform {
```

Comment:
This name may be a bit confusing.  It implies an array of objects is the type, e.g. `[{}, {}]`.  But it's more of an object _to_ array transform.